x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/planner/Translatable.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ql.planner;
+
+import org.elasticsearch.xpack.ql.querydsl.query.Query;
+
+import java.time.ZoneId;
+
+public interface Translatable {
+    Query getQuery(String name, Object value, String format, boolean isDateLiteralComparison, ZoneId zoneId);
```

Comment:
That seems reasonable to me.  It's definitely a cleaner function signature; in fact, we probably don't even want to pass in the `BinaryComparison`, since that's what will be implementing the interface.

If we go this route, I'd move the type coercion logic to its own static method, to avoid too much duplication.  Having an easy way to override that for ES|QL sounds nice though.

Reply:
Err, I just realized a big copy-paste mistake; I meant to propose the interface
```
Query translate(TranslatorHandler handler)
```